prototype_source/maskedtensor_adagrad.rst

```diff
@@ -0,0 +1,182 @@
+Efficiency of writing "sparse" semantics for Adagrad
```

Comment:
The title here leaves the reader unsure what to expect--is this offering an alternative? What is the alternative if so?

Maybe something clearer can be: 
```suggestion
Efficiently writing "sparse" semantics for Adagrad with MaskedTensor
```

Reply:
good point!